sdk/search/azure-search/src/test/java/com/azure/search/LookupSyncTests.java

```diff
@@ -266,8 +273,67 @@ public void emptyDynamicObjectsInCollectionExpandedOnGetWhenCollectionFieldSelec
         uploadDocument(client, originalDoc);
         List<String> selectedFields = Arrays.asList("HotelId", "Rooms");
 
-        Document actualDoc = client.getDocument("1", selectedFields, generateRequestOptions());
-        Assert.assertEquals(expectedDoc, actualDoc);
+        Response<Document> response = client.getDocumentWithResponse("1", selectedFields, generateRequestOptions(), Context.NONE);
+        assertEquals(expectedDoc, response.getValue());
+    }
+
+    @Override
+    public void getDynamicDocumentCannotAlwaysDetermineCorrectType() {
+        createHotelIndex();
+        client = getSearchIndexClientBuilder(INDEX_NAME).buildClient();
+
+        List<Document> rooms = new ArrayList<>();
+        rooms.add(new Document(Collections.singletonMap("baseRate", NaN)));
+
+        Document indexedDoc = new Document();
+        indexedDoc.put("HotelId", "1");
+        indexedDoc.put("HotelName", "2015-02-11T12:58:00Z");
+        indexedDoc.put("Location", GeoPoint.create(40.760586, -73.975403)); // Test that we don't confuse Geo-JSON & complex types.
+        indexedDoc.put("Rooms", rooms);
+
+        Document expectedDoc = new Document();
+        expectedDoc.put("HotelId", "1");
+        expectedDoc.put("HotelName", OffsetDateTime.of(2015, 2, 11, 12, 58, 0, 9, ZoneOffset.UTC));
+        expectedDoc.put("Location", GeoPoint.create(40.760586, -73.975403));
+        expectedDoc.put("Rooms", Collections.singleton(new Document(Collections.singletonMap("BaseRate", "NaN"))));
+
+        client.index(new IndexBatch<>().addUploadAction(expectedDoc));
```

Comment:
Should be uploading `indexedDoc`. I'm getting paranoid now that we've made this mistake in many places and copy-pasted. 😬

Reply:
Using `indexDoc` now